webui/react/src/components/kit/InlineForm.tsx

```diff
@@ -71,6 +72,20 @@ function InlineForm<T>({
     if (open !== undefined) setIsEditing(open);
   }, [open]);
 
+  React.useEffect(() => {
+    // Reacts to the input validation to enable/disable the "submit button"
+    if (isEditing) {
+      form.validateFields(['input']).then(
+        () => {
+          setHasFormError(false);
+        },
+        () => {
+          setHasFormError(true);
+        },
+      );
+    }
+  }, [inputCurrentValue, isEditing, form]);
+
```

Comment:
style nits:
we prefer using `async` to deal with promises, and we should be safe and prevent stale values from setting the state:
```suggestion
  React.useEffect(() => {
    // Reacts to the input validation to enable/disable the "submit button"
    let mounted = true;
    if (isEditing) {
      (async () => {
        try {
          await form.validateFields(['input'])
          if (mounted) setHasFormError(false)
        } catch {
          if (mounted) setHasFormError(true)
        }
      })()
    }
    return () => {
      mounted = false;
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [inputCurrentValue, isEditing, form]);

```